packages/builder/src/components/userInterface/EventsEditor/actions/SaveRecord.svelte

```diff
@@ -0,0 +1,138 @@
+<script>
+  import { Select, Label } from "@budibase/bbui"
+  import { store, backendUiStore } from "builderStore"
+  import fetchBindableProperties from "builderStore/fetchBindableProperties"
+  import SaveFields from "./SaveFields.svelte"
+  import {
+    readableToRuntimeBinding,
+    runtimeToReadableBinding,
+  } from "builderStore/replaceBindings"
+
+  // parameters.contextPath used in the client handler to determine which record to save
+  // this could be "data" or "data.parent", "data.parent.parent" etc
+  export let parameters
+
+  $: bindableProperties = fetchBindableProperties({
+    componentInstanceId: $store.currentComponentInfo._id,
+    components: $store.components,
+    screen: $store.currentPreviewItem,
+    models: $backendUiStore.models,
+  })
+
+  // we pick all the _id fields, to determine all the available
+  // record contexts
+  let idFields
+
+  const idBindingToContextPath = id => id.substring(0, id.length - 4)
+  const contextPathToId = path => `${path}._id`
+
+  $: {
+    idFields = bindableProperties.filter(
+      bindable =>
+        bindable.type === "context" && bindable.runtimeBinding.endsWith("._id")
+    )
+    // ensure contextPath is always defaulted - there is usually only one option
+    if (idFields.length > 0 && !parameters.contextPath) {
+      parameters.contextPath = idBindingToContextPath(
+        idFields[0].runtimeBinding
+      )
+      parameters = parameters
+    }
+  }
+
+  // just wraps binding in {{ ... }}
+  const toBindingExpression = bindingPath => `{{ ${bindingPath} }}`
+
+  // finds the selected idBinding, then reads the table/view
+  // from the component instance that it belongs to.
+  // then returns the field names for that schema
+  const schemaFromContextPath = contextPath => {
+    if (!contextPath) return []
+
+    const idBinding = bindableProperties.find(
+      prop => prop.runtimeBinding === contextPathToId(contextPath)
+    )
+    if (!idBinding) return []
+
+    const { instance } = idBinding
+
+    const component = $store.components[instance._component]
+
+    // component.context is the name of the prop that holds the modelId
+    const modelInfo = instance[component.context]
+    const modelId =
+      typeof modelInfo === "string" ? modelInfo : modelInfo.modelId
+
+    if (!modelInfo) return []
+
+    const model = $backendUiStore.models.find(m => m._id === modelId)
+    parameters.modelId = modelId
+    return Object.keys(model.schema).map(k => ({
+      name: k,
+      type: model.schema[k].type,
+    }))
+  }
+
+  let schemaFields
+  $: {
```

Comment:
NAB: Most of the other svelte components in budibase follow this convention:

```
<script>
// constants

// props

// state

// reactive declarations

// functions 
</script>
```

It's a little easier to read this way, compared to jumping around.